projects/distributed-tracing/src/shared/dashboard/widgets/table/table-widget.model.ts

```diff
@@ -71,6 +93,20 @@ export class TableWidgetModel {
   })
   public mode: TableMode = TableMode.Flat;
 
+  @ModelProperty({
+    key: 'modeToggles',
+    displayName: 'Modes Toggle',
+    // tslint:disable-next-line: no-object-literal-type-assertion
+    type: {
+      key: ARRAY_PROPERTY.type,
+      subtype: {
+        key: ENUM_TYPE.type,
+        values: [TableMode.Flat, TableMode.Tree, TableMode.Detail]
+      }
+    } as ArrayPropertyTypeInstance
+  })
+  public modeToggles: TableMode[] = [];
```

Comment:
this feels like it conflicts with mode - do we think it's feasible to unify the two properties? something like `modes` which accepts an array and defaults to a single mode?

Reply:
I don't think it should conflict. The `mode` property is adequate for 99% of our use cases so I would prefer not to make users build out an array and object when unnecessary.

Reply:
OK fair. I guess with an array like this, you'd want a way to specify the default anyway. Thinking about it that way, maybe `modeOptions` is a better name (feel free to ignore if you're not back in here).

Reply:
I like that name better too. I'll change when I'm back in there. Same for filterOptions.